webapp/javascript/components/Settings/Apps/index.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useEffect, useState } from 'react';
+import cl from 'classnames';
+import { useAppDispatch, useAppSelector } from '@webapp/redux/hooks';
+import {
+  selectApps,
+  reloadApps,
+  deleteApp,
+  selectIsLoadingApps,
+} from '@webapp/redux/reducers/settings';
+import { addNotification } from '@webapp/redux/reducers/notifications';
+import { type App } from '@webapp/models/app';
+import Input from '@webapp/ui/Input';
+import TableUI from '@webapp/ui/Table';
+import LoadingSpinner from '@webapp/ui/LoadingSpinner';
+import { getAppTableRows } from './getAppTableRows';
+
+import appsStyles from './Apps.module.css';
+import tableStyles from '../SettingsTable.module.scss';
+
+const headRow = [
+  { name: '', label: 'Name', sortable: 0 },
+  { name: '', label: '', sortable: 0 },
+];
+
+function Apps() {
+  const dispatch = useAppDispatch();
+  const apps = useAppSelector(selectApps);
+  const isLoading = useAppSelector(selectIsLoadingApps);
+  const [search, setSearchField] = useState('');
+  const [appsInProcessing, setAppsInProcessing] = useState([] as string[]);
+  const [deletedApps, setDeletedApps] = useState([] as string[]);
+
+  useEffect(() => {
+    dispatch(reloadApps());
+  }, []);
+
+  const displayApps =
+    (apps &&
+      apps.filter(
+        (x) =>
+          x.name.toLowerCase().indexOf(search.toLowerCase()) !== -1 &&
+          !deletedApps.includes(x.name)
+      )) ||
+    [];
+
+  const handleDeleteApp = (app: App) => {
+    setAppsInProcessing([...appsInProcessing, app.name]);
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    dispatch(deleteApp(app))
+      .unwrap()
+      .then(() => {
+        setAppsInProcessing(appsInProcessing.filter((x) => x !== app.name));
+        setDeletedApps([...deletedApps, app.name]);
+        // eslint-disable-next-line @typescript-eslint/no-floating-promises
```

Comment:
Maybe it's better to move same rules to top of the file (line 48, 54 etc.)?

`/* eslint-disable @typescript-eslint/no-floating-promises */
`

Reply:
fixed